lib/chef/provider/route.rb

```diff
@@ -154,58 +154,57 @@ def action_delete
         end
 
         # for now we always write the file (ugly but its what it is)
-        generate_config
+        write_config
       end
 
-      def generate_config
+      def write_config
+        return unless platform_family?("rhel", "fedora", "amazon")
+
         conf = {}
-        case node[:platform]
-        when "centos", "redhat", "fedora"
```

Comment:
kind of like the case statement here better than the early return, its more indicative that this is where you'd need to add support for debian and other linuxes...